kernel-util/src/main/java/org/sakaiproject/content/util/ZipContentUtil.java

```diff
@@ -394,8 +401,36 @@ private void storeContentCollection(String rootId, ContentCollection collection,
 	 * @param out
 	 * @throws Exception
 	 */
-	private void storeContentResource(String rootId, ContentResource resource, ZipOutputStream out) throws Exception {		
-		String filename = resource.getId().substring(rootId.length(),resource.getId().length());				
+	private void storeContentResource(String rootId, ContentResource resource, ZipOutputStream out) throws Exception {
+		String filename = resource.getId().substring(rootId.length(),resource.getId().length());
+
+		//Inorder to have username as the folder name rather than having eids
+		if(filename != null && filename.length()>0) {
+			String filenameArr[] = filename.split(Entity.SEPARATOR);
+			if (filenameArr.length > 1) {
+				ContentCollectionEdit collectionEdit = (ContentCollectionEdit) ContentHostingService.getCollection(rootId+filenameArr[0]+Entity.SEPARATOR);
+				ResourcePropertiesEdit props = collectionEdit.getPropertiesEdit();
+				String displayName = props.getProperty(ResourcePropertiesEdit.PROP_DISPLAY_NAME);
+				try {
+					String uniqueId = UserDirectoryService.getUser(filenameArr[0]).getDisplayId();
```

Comment:
Does this end up doing an user lookup for every file included in the export? Although there is heavy caching of found users this will query the DB for every file I think?